lib/features.js

```diff
@@ -64,6 +64,40 @@ module.exports = config => {
       return isSampledUser(signinConfirmation.sample_rate, uid, 'signinConfirmation')
     },
 
+
+    /**
+     * Returns whether or not to use signin unblock feature on a request.
+     *
+     * @param account
+     * @param config
+     * @param request
+     * @returns {boolean}
+     */
+    isSigninUnblockEnabledForUser(uid, email, request) {
+      if (! signinUnblock.enabled) {
+        return false
+      }
+
+      // Or if the email address matching one of these regexes.
+      if (signinUnblock.allowedEmailAddresses.test(email)) {
+        return true
+      }
+
+      // While we're testing this feature, there may be some funky
+      // edge-cases in device login flows that haven't been fully tested.
+      // Temporarily avoid them for regular users by checking the `context` flag,
+      const context = request.payload &&
+        request.payload.metricsContext &&
+        request.payload.metricsContext.context
+
+      if (signinUnblock.supportedClients.indexOf(context) === -1) {
```

Comment:
We decided the content server is always going to send a metricsContext.context to the auth server. New values in https://github.com/mozilla/fxa-content-server/pull/4251 are `web` and `oauth`. My patch here is unnecessary.